### ISSUE TITLE
Getting error when using combineTypedReducers

### ISSUE THREAD

**Issue by jonstodle** [brianegan/flutter_redux#33]
Given

```
final store = new Store<int>(appReducer, initialState: 0);

final appReducer = combineTypedReducers(
    [new ReducerBinding<int, Increment>((state, action) => state + 1)]);

class Increment{}

@override
  Widget build(BuildContext context) {
    return new StoreProvider<int>(
        store: store,
        child: new MaterialApp(
          title: 'Flutter Demo',
          theme: new ThemeData(
            primarySwatch: Colors.blue,
          ),
          home: new MyHomePage(title: 'Flutter Demo Home Page'),
        ));
  }
```

I get the following error when trying to dispatch an `Increment` action:

```
type '(int, Increment) => int' is not a subtype of type '(int, dynamic) => int'
```

This is when running `flutter run --preview-dart-2`. When running `flutter run`, I get the follwing error:

```
The following StoreProviderError was thrown building StoreConnector<int, Store<int>>(dirty):
Error: No dynamic found. To fix, please try:

* Using Dart 2 (required) by using the --preview-dart-2 flag
* Wrapping your MaterialApp with the StoreProvider<State>, rather than an individual Route
* Providing full type information to your Store<State>, StoreProvider<State> and
StoreConnector<State, ViewModel>
```

```
flutter --version
Flutter 0.2.3 • channel beta • https://github.com/flutter/flutter.git
Framework • revision 5a58b36e36 (2 weeks ago) • 2018-03-13 13:20:13 -0700
Engine • revision e61bb9ac3a
Tools • Dart 2.0.0-dev.35.flutter-290c576264
```

Repro project can be found here: https://github.com/jonstodle/flutter_redux_reproduction

**Comment by brianegan** [comment]
Hey hey :) Sorry about the trouble! That issue is fixed with the latest version of Redux: 3.0.0.

  * Migration Guide: https://github.com/johnpryan/redux.dart/#redux-300-migration--dart-2-support
  * You'll also need to update to flutter_redux 0.5.0

**Comment by jonstodle** [comment]
Brilliant! Thanks
